### PR TITLE
Replaced twitter JSON parser with Jackson JSON parser

### PR DESCRIPTION
**Pull request by liming-thu**
The twitter JSON parser is the biggest bottleneck currently, because it parsed all the fields of tweets, the Jackson JSON parser allows to read specific fields, so I used the Jackson JSON parser to replace it.
I rewrote the whole process of fields retrieving and adm constructing. Now, the speed increases from 4000/s to 6000/s.
I used 3 million tweets to test them, the output is the same as the current except the date/time part.
The string->date->string process is too expensive, so I use the parse_date/time() instead of date(), but didn't test on AsterixDB.

### CODE REVIEW

**Comment by chenlica** [comment]
@baiqiushi : please review it.

File: examples/twittermap/noah/src/test/scala/edu/uci/ics/cloudberry/noah/TwitterJSONTagToADMTest.scala [hunk 1]
```diff
@@ -9,9 +9,9 @@ import edu.uci.ics.cloudberry.util.Profile._
 class TwitterJSONTagToADMTest extends FlatSpec with Matchers {
   "USGeoGnosis" should "tag the us json file" in {
     val sLoadGeo = new Date()
-    val shapeMap = Map.apply(StateLevel -> new File("web/public/data/state.json"),
-      CountyLevel -> new File("web/public/data/county.json"),
-      CityLevel -> new File("web/public/data/city.json"))
+    val shapeMap = Map.apply(StateLevel -> new File("/root/cloudberry/examples/twittermap/web/public/data/state.json"),
+      CountyLevel -> new File("/root/cloudberry/examples/twittermap/web/public/data/county.json"),
+      CityLevel -> new File("/root/cloudberry/examples/twittermap/web/public/data/city.json"))
```
Comment (waans11) [inline examples/twittermap/noah/src/test/scala/edu/uci/ics/cloudberry/noah/TwitterJSONTagToADMTest.scala]: Why is this change necessary?

File: examples/twittermap/noah/src/main/scala/edu/uci/ics/cloudberry/noah/TwitterJSONTagToADM.scala [hunk 2]
```diff
@@ -52,7 +54,6 @@ object TwitterJSONTagToADM {
       }
     }
   }
```
Comment (waans11) [inline examples/twittermap/noah/src/main/scala/edu/uci/ics/cloudberry/noah/TwitterJSONTagToADM.scala]: Let's not make any unnecessary changes. Line 55 was removed. (doesn't have to be)

File: examples/twittermap/noah/src/main/scala/edu/uci/ics/cloudberry/noah/adm/ADM.java [hunk 3]
```diff
@@ -17,30 +15,46 @@
     public static final SimpleDateFormat ADMTimeFormat = new SimpleDateFormat("HH:mm:ss.SSS");
 
     public static String mkADMConstructor(String constructor, String content) {
-        return constructor + "(\"" + content + "\")";
+        StringBuilder sbConstructor=new StringBuilder();
```
Comment (waans11) [inline examples/twittermap/noah/src/main/scala/edu/uci/ics/cloudberry/noah/adm/ADM.java]: StringBuilder sbConstructor = new StringBuilder();

// Can you apply the correct source format? There was no space in the line.

Comment (waans11) [reply examples/twittermap/noah/src/main/scala/edu/uci/ics/cloudberry/noah/adm/ADM.java]: And why this string builder is necessary?

File: examples/twittermap/noah/src/main/scala/edu/uci/ics/cloudberry/noah/adm/ADM.java [hunk 4]
```diff
@@ -17,30 +15,46 @@
...
+        sbConstructor.append(constructor).append("(\"").append(content).append("\")");
+        return sbConstructor.toString();
     }
 
     public static String mkInt64Constructor(long value) {
         return mkADMConstructor("int64", String.valueOf(value));
     }
+
```
Comment (waans11) [inline examples/twittermap/noah/src/main/scala/edu/uci/ics/cloudberry/noah/adm/ADM.java]: Let's not make unnecessary format changes.

File: examples/twittermap/noah/src/main/scala/edu/uci/ics/cloudberry/noah/adm/ADM.java [hunk 5]
```diff
@@ -17,30 +15,46 @@
...
     public static String mkInt8Constructor(String value) {
         return mkADMConstructor("int8", value);
     }
+
```
Comment (waans11) [inline examples/twittermap/noah/src/main/scala/edu/uci/ics/cloudberry/noah/adm/ADM.java]: Let's not make unnecessary format changes.

File: examples/twittermap/noah/src/main/scala/edu/uci/ics/cloudberry/noah/adm/ADM.java [hunk 6]
```diff
@@ -17,30 +15,46 @@
...
     public static String mkInt32Constructor(String value) {
         return mkADMConstructor("int32", value);
     }
+
```
Comment (waans11) [inline examples/twittermap/noah/src/main/scala/edu/uci/ics/cloudberry/noah/adm/ADM.java]: Let's not make unnecessary format changes.

File: examples/twittermap/noah/src/main/scala/edu/uci/ics/cloudberry/noah/adm/ADM.java [hunk 7]
```diff
@@ -17,30 +15,46 @@
...
     public static String mkDateTimeConstructor(Date jdate) {
         return "datetime(\"" + ADMDateFormat.format(jdate) + "T" + ADMTimeFormat.format(jdate) + "\")";
     }
+    public static String mkDateTimeStringFromTweet(String srcDateTimeString) {
+        if(srcDateTimeString.equals(""))
+            return null;
+        StringBuilder sbDateTime=new StringBuilder();
+        sbDateTime.append("parse_datetime(\"").append(srcDateTimeString).append("\",\"W MMM DD hh:mm:ss z YYYY\")");
+        return sbDateTime.toString();
+    }
```
Comment (waans11) [inline examples/twittermap/noah/src/main/scala/edu/uci/ics/cloudberry/noah/adm/ADM.java]: Why is StringBuilder necessary? Also, can you put a comment about this function before the function?

File: examples/twittermap/noah/src/main/scala/edu/uci/ics/cloudberry/noah/adm/ADM.java [hunk 8]
```diff
@@ -17,30 +15,46 @@
...
+
+    public static String mkDateStringFromTweet(String srcDateTimeString) {
+        if (srcDateTimeString.equals(""))
+            return null;
+        StringBuilder sbDate=new StringBuilder();
```
Comment (waans11) [inline examples/twittermap/noah/src/main/scala/edu/uci/ics/cloudberry/noah/adm/ADM.java]: Why is StringBuilder necessary?

File: examples/twittermap/noah/src/main/scala/edu/uci/ics/cloudberry/noah/adm/ADM.java [hunk 9]
```diff
@@ -97,43 +96,17 @@ public static void keyValueToSbWithComma(StringBuilder sb, String key, String va
...
-        for (int i = 0; i < hashtagEntities.length; i++) {
-            if (i > 0) {
-                sb.append(',');
-            }
-            sb.append(mkQuote(hashtagEntities[i].getText()));
-        }
-        sb.append("}}");
-        return sb.toString();
+    public static String mkQuote(String str) {// quote and escape
+        StringBuilder sbQuote=new StringBuilder();
```
Comment (waans11) [inline examples/twittermap/noah/src/main/scala/edu/uci/ics/cloudberry/noah/adm/ADM.java]: StringBuilder sbQuote = new StringBuilder();

File: examples/twittermap/noah/src/main/scala/edu/uci/ics/cloudberry/noah/adm/ADM.java [hunk 10]
```diff
@@ -97,43 +96,17 @@ public static void keyValueToSbWithComma(StringBuilder sb, String key, String va
...
-        sb.append("}}");
-        return sb.toString();
-    }
-
-    public static String mkQuote(String str) {
-        StringBuilder sb = new StringBuilder();
-        sb.append('"').append(StringEscapeUtils.escapeJava(str)).append('"');
-        return sb.toString();
+    public static String mkQuoteOnly(String str){
+        StringBuilder sbQuote=new StringBuilder();
```
Comment (waans11) [inline examples/twittermap/noah/src/main/scala/edu/uci/ics/cloudberry/noah/adm/ADM.java]: StringBuilder sbQuote = new StringBuilder();

File: examples/twittermap/noah/src/main/scala/edu/uci/ics/cloudberry/noah/adm/MyTweet/MyPlace.java [hunk 11]
```diff
@@ -0,0 +1,159 @@
...
+        this.name = name;
+    }
+
+    @Override
+    public String toString() {
+        return full_name;
+    }
+
+    public String toADM() {
+        StringBuilder placeSB=new StringBuilder();
```
Comment (waans11) [inline examples/twittermap/noah/src/main/scala/edu/uci/ics/cloudberry/noah/adm/MyTweet/MyPlace.java]: StringBuilder placeSB = new StringBuilder();

File: examples/twittermap/noah/src/main/scala/edu/uci/ics/cloudberry/noah/feed/TagTweet.scala [hunk 12]
```diff
@@ -19,7 +19,7 @@ object TagTweet {
   @throws[UnknownPlaceException]
   @throws[TwitterException]
   def tagOneTweet(ln: String, requireGeoField: Boolean): String = {
-    val adm = Tweet.toADM(TwitterObjectFactory.createStatus(ln), usGeoGnosis, requireGeoField)
+    val adm = Tweet.toADM(ln, usGeoGnosis, requireGeoField)
```
Comment (waans11) [inline examples/twittermap/noah/src/main/scala/edu/uci/ics/cloudberry/noah/feed/TagTweet.scala]: Why this change is necessary?

Comment (liming-thu) [reply examples/twittermap/noah/src/main/scala/edu/uci/ics/cloudberry/noah/feed/TagTweet.scala]: The "TwitterObjectFactory.createStatus" using Twitter JSON parser, we replaced it, so just need pass the string line to "toADM".

File: examples/twittermap/noah/src/main/scala/edu/uci/ics/cloudberry/noah/adm/ADM.java (hunk 8 above)
Comment (liming-thu) [reply examples/twittermap/noah/src/main/scala/edu/uci/ics/cloudberry/noah/adm/ADM.java]: "StringBuilder.append" is the fastest way to construct a new string, so I replaced the string concat operation with StringBuilder.

File: examples/twittermap/noah/src/main/scala/edu/uci/ics/cloudberry/noah/adm/ADM.java (hunk 6 above)
Comment (liming-thu) [reply examples/twittermap/noah/src/main/scala/edu/uci/ics/cloudberry/noah/adm/ADM.java]: This function is from Jianfeng, it just add the "int32" prefix to the "String value" without changing the format, the return value is "int32( value )".

File: examples/twittermap/noah/src/test/scala/edu/uci/ics/cloudberry/noah/TwitterJSONTagToADMTest.scala (hunk 1 above)
Comment (liming-thu) [reply examples/twittermap/noah/src/test/scala/edu/uci/ics/cloudberry/noah/TwitterJSONTagToADMTest.scala]: The Jackson JSON Parser need to load the Class Object to match the JSON object while parsing JSON string, so I created a new MyPlace.java. Since we don't use Twitter JSON parser anymore, the Place.java is also unnecessary.
"/root/cloudberry/...." is unnecessary, just used in my test environment, sorry to forgot to delete them.
